plonky2x/src/frontend/hash/sha/sha256_curta.rs

```diff
@@ -131,26 +132,54 @@ impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
     pub fn curta_sha256(&mut self, input: &[ByteVariable]) -> Bytes32Variable {
         let padded_input = self.curta_sha256_pad(input);
 
-        let bytes = self.bytes_to_target(&padded_input);
+        if self.sha256_accelerator.is_none() {
+            self.sha256_accelerator = Some(Sha256Accelerator::<L, D> {
+                sha256_requests: Vec::new(),
+                sha256_responses: Vec::new(),
+                _marker: PhantomData,
+            });
+        }
+
+        let accelerator = self
+            .sha256_accelerator
+            .as_mut()
+            .expect("sha256 accelerator should exist");
+        let bytes = padded_input
+            .iter()
+            .map(|x| convert_byte_var_to_target(*x, &mut self.api))
+            .collect::<Vec<_>>();
 
-        self.sha256_requests.push(bytes);
+        accelerator.sha256_requests.push(bytes);
         let digest = self.api.add_virtual_target_arr::<32>();
-        self.sha256_responses.push(digest);
-        Bytes32Variable::from_targets(
-            &digest
-                .into_iter()
-                .flat_map(|byte| {
-                    let mut bits = self
-                        .api
-                        .low_bits(byte, 8, 8)
-                        .into_iter()
-                        .map(|b| b.target)
-                        .collect_vec();
-                    bits.reverse();
-                    bits
-                })
-                .collect_vec(),
-        )
+        accelerator.sha256_responses.push(digest);
+
+        let bytes: [ByteVariable; 32] =
+            digest.map(|x| convert_byte_target_to_byte_var(x, &mut self.api));
+        bytes.into()
+    }
+
+    /// Executes a SHA256 hash on the given input with a mutable accelerator.
+    /// Note: Should only be called by curta_constrain_sha256.
```

Comment:
nit: https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md#appendix-a-full-conventions-text

there should be a line break after 161 to seperate the main description of a function from its details